File: gameserver/src/main/java/l2s/gameserver/network/authcomm/gs2as/PlayerAuthRequest.java
package l2s.gameserver.network.authcomm.gs2as;

import io.netty.buffer.ByteBuf;
import l2s.gameserver.network.authcomm.SendablePacket;
import l2s.gameserver.network.l2.GameClient;

public class PlayerAuthRequest extends SendablePacket {
    private final String account;
    private final int playOkID1;
    private final int playOkID2;
    private final int loginOkID1;
    private final int loginOkID2;

    public PlayerAuthRequest(GameClient client) {
        account = client.getLogin();
        playOkID1 = client.getSessionKey().playOkID1;
        playOkID2 = client.getSessionKey().playOkID2;
        loginOkID1 = client.getSessionKey().loginOkID1;
        loginOkID2 = client.getSessionKey().loginOkID2;
    }

    @Override
    protected ByteBuf writeImpl(ByteBuf byteBuf) {
        writeString(byteBuf, account);
        byteBuf.writeInt(playOkID1);
        byteBuf.writeInt(playOkID2);
        byteBuf.writeInt(loginOkID1);
        byteBuf.writeInt(loginOkID2);
        return byteBuf;
    }

    @Override
    protected byte getOpCode() {
        return 2;
    }
}

File: gameserver/src/main/java/l2s/gameserver/network/authcomm/gs2as/PlayerAuthRequest.kt
package l2s.gameserver.network.authcomm.gs2as

import io.netty.buffer.ByteBuf
import l2s.gameserver.network.authcomm.SendablePacket
import l2s.gameserver.network.l2.GameClient

class PlayerAuthRequest(client: GameClient) : SendablePacket() {
    private val account: String = client.login
    private val playOkID1: Int = client.sessionKey.playOkID1
    private val playOkID2: Int = client.sessionKey.playOkID2
    private val loginOkID1: Int = client.sessionKey.loginOkID1
    private val loginOkID2: Int = client.sessionKey.loginOkID2

    override val opCode: Byte = 0x02

    override fun writeImpl(byteBuf: ByteBuf): ByteBuf {
        writeString(byteBuf, account)
        byteBuf.writeInt(playOkID1)
        byteBuf.writeInt(playOkID2)
        byteBuf.writeInt(loginOkID1)
        byteBuf.writeInt(loginOkID2)
        return byteBuf
    }
}